### DISCUSSION TITLE
Is there any example DDD code for the Issue example?

### DISCUSSION THREAD

**Discussion by sturlath** [abpframework/abp#8704] [Q&A] [answered]
I have been reading the [Domain-Driven-Design-Implementation-Guide](https://docs.abp.io/en/abp/4.2/Domain-Driven-Design-Implementation-Guide) (in effort to learn how to apply best practice DDD to the abp framework) and see lot of things I would like to explore further with full context in code, so my question is if there is a repository for the  [example domain (issue)](https://docs.abp.io/en/abp/4.2/Domain-Driven-Design-Implementation-Guide#the-example-domain)?

It would be really valuable to get to see the whole DDD picture in combo with the framework. 

I also can't see that any of the [samples](https://docs.abp.io/en/abp/latest/Samples/Index) is a good match... I might be wrong..

**Comment by ebicoglu** [comment, marked as answer]
check out https://docs.abp.io/en/commercial/latest/samples/easy-crm

**Reply by sturlath** [reply]
I'm getting "500 An internal error occurred during your request!" on https://abp.io/api/download/samples/easy-crm

**Reply by maliming** [reply]
hi @sturlath I will check this.

**Reply by elcoda** [reply]
Hi, 
are you sure we can consider easy-crm an example of DDD pattern applied via ABP?

Reading ABP Book (Mastering ABP Framework) and checking easy-crm I expect, for example, that OrderLineAppService.CreateAsync(...) won't use directly OrderLines and add them to the order.OrderLines.Add(...). 

I would expect a method on Order class that adds lines to orders (for example AddLineToOrder(...)) and allows AppServices to use only Order as aggregate root without touching OrderLine model.

Am I right?

Many thanks

**Reply by elcoda** [reply]
For further reading: https://docs.abp.io/en/abp/latest/Entities contains answer to my question.
easy-crm seems not using DDD approach but in the Abp docs we have an example of implementation of "AddProduct"
